app/controllers/users/passwords_controller.rb

```diff
@@ -2,19 +2,29 @@ module Users
   class PasswordsController < Devise::PasswordsController
     include ValidEmailParameter
 
-    before_action :confirm_valid_token, only: [:edit]
-
     def create
-      RequestPasswordReset.new(params[:user][:email]).perform
+      email = params[:user][:email]
+
+      RequestPasswordReset.new(email).perform
+
+      analytics_user = User.find_by_email(email) || AnonymousUser.new
```

Comment:
Hmm.. I hope this doesn't start another long debate, but I'm not sure that anonymous users is the right thing to report here.   If nobody is logged in then the `current_user` is the anonymous user.   So the anonymous user might be the one who is requesting the reset, but user that the anonymous user want to reset the password for is the one that matches the mail.  When no user is found matching that email we want to return some kind of *missing user*.   Think about the following two sentences:

`An attempt was made to reset the password for the anonymous user`

`An attempt was made to reset the password for a non-existent user`

The latter is an accurate description of what is happening.  The former doesn't mean any thing as far as i can tell.

So I'm sad to say that I think this is another case when the result of the DB lookup should be checked and handled explicitly.    (or at least change the class to MissingUser)

One thing that makes MissingUser different from AnonymousUser, for example, is that MissingUser.email can/should be a real email address.   However I think MissingUser is probably overkill in this case.